frontend/javascripts/dashboard/explorative_annotations_view.js

```diff
@@ -96,15 +97,21 @@ class ExplorativeAnnotationsView extends React.PureComponent<Props, State> {
     searchQuery: "",
     tags: [],
     isLoading: false,
+    performedInitialFetch: false,
   };
 
   componentDidMount() {
     this.setState(persistence.load(this.props.history));
-    this.fetchNextPage(0);
```

Comment:
Cool that you found the issue :+1: However, I think, that there is another race condition looming: `fetchNextPage` itself fetches annotations (depending on `this.state.shouldShowArchivedTracings`) and then calls `setModeState` which itself reads again from `this.state.shouldShowArchivedTracings`. If the user toggles the state after the annotations-fetch has started but before it finished, the annotations will be pushed into the wrong array. I think, it would make sense to adapt `setModeState` to accept the `shouldShowArchivedTracings` parameter directly (instead of reading it out again). Do you agree?

Reply:
> Do you agree?

Jep 👍 , that sounds like a reasonable approach to fix this.